File: app/src/main/java/com/gu/swiperefreshplus/fragment/DemoContact.java
package com.gu.swiperefreshplus.fragment;

import java.util.List;

/**
 * Created by gu on 2016/12/22.
 */

public class DemoContact {
    interface View{
        void onDataChange();
        void onDataAdded(int from,int to);
        void setPresenter(Presenter presenter);
    }
    interface Presenter{
        void refresh();
        void loadMore();
        List getData();
        void bind();
        void unbind();
    }
}

File: app/src/main/java/com/gu/swiperefreshplus/fragment/DemoContact.kt
package com.gu.swiperefreshplus.fragment;


/**
 * Created by gu on 2016/12/22.
 */

 class DemoContact {
    interface View{
        fun onDataChange();
        fun onDataAdded(from:Int,to:Int);
        fun setPresenter(presenter:Presenter );
    }
    interface Presenter{
        fun refresh();
        fun loadMore();
        fun getData():List<Int>;
        fun bind();
        fun unbind();
    }
}
